src/rdkafka_mock_cgrp.c

```diff
@@ -706,3 +721,930 @@ void rd_kafka_mock_cgrps_connection_closed(rd_kafka_mock_cluster_t *mcluster,
+                return;
+        }
+
+        mcgrp->manual_assignment = rd_true;
+
+        rd_kafka_mock_cgrp_consumer_target_assignment_set(mcgrp,
+                                                          target_assignment);
+}
+
+/**
+ * @brief Sets \p member current assignment to a copy of
+ *        \p current_assignment.
+ *
+ * @param member A consumer group member.
+ * @param current_assignment Current assignment to set.
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+static void rd_kafka_mock_cgrp_consumer_member_current_assignment_set(
+    rd_kafka_mock_cgrp_consumer_member_t *member,
+    const rd_kafka_topic_partition_list_t *current_assignment) {
+        if (member->current_assignment) {
+                rd_kafka_topic_partition_list_destroy(
+                    member->current_assignment);
+        }
+
+        member->current_assignment =
+            current_assignment
+                ? rd_kafka_topic_partition_list_copy(current_assignment)
+                : NULL;
+}
+
+/**
+ * @brief Sets \p member returned assignment to a
+ *        copy of \p returned_assignment.
+ *
+ * @param member A consumer group member.
+ * @param returned_assignment Returned assignment to set.
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+static void rd_kafka_mock_cgrp_consumer_member_returned_assignment_set(
+    rd_kafka_mock_cgrp_consumer_member_t *member,
+    const rd_kafka_topic_partition_list_t *returned_assignment) {
+        if (member->returned_assignment) {
+                rd_kafka_topic_partition_list_destroy(
+                    member->returned_assignment);
+        }
+        member->returned_assignment =
+            returned_assignment
+                ? rd_kafka_topic_partition_list_copy(returned_assignment)
+                : NULL;
+}
+
+/**
+ * @brief Returns a copy of \p assignment containing only partitions
+ *        that can be assignment, whose topic id is non-zero.
+ *
+ * @param assignment The assignment to filter.
+ * @return Filtered assignment.
+ *
+ * @remark The returned pointer ownership is transferred to the caller.
+ */
+static rd_kafka_topic_partition_list_t *
+rd_kafka_mock_cgrp_consumer_member_assignment_filter(
+    rd_kafka_topic_partition_list_t *assignment) {
+        rd_kafka_topic_partition_t *rktpar;
+        rd_kafka_topic_partition_list_t *ret =
+            rd_kafka_topic_partition_list_new(assignment->cnt);
+
+        RD_KAFKA_TPLIST_FOREACH(rktpar, assignment) {
+                rd_kafka_Uuid_t topic_id =
+                    rd_kafka_topic_partition_get_topic_id(rktpar);
+                if (rd_kafka_Uuid_cmp(topic_id, RD_KAFKA_UUID_ZERO)) {
+                        rd_kafka_topic_partition_list_add_copy(ret, rktpar);
+                }
+        }
+
+        return ret;
+}
+
+/**
+ * Returns true iff \p new_assignment doesn't have any intersection with any
+ * other member current assignment.
+ */
+rd_bool_t rd_kafka_mock_cgrp_consumer_member_next_assignment_can_bump_epoch(
```

Comment:
I can add this statement:
```
If there's an intersection it means we cannot bump the epoch at the moment because some
of these partitions are held by a different member, they have to be released from that member
before it's possible to increase the epoch and assign additional partitions to this member.
```